flink-yarn/src/main/scala/org/apache/flink/yarn/ApplicationMasterBase.scala

```diff
@@ -120,16 +122,73 @@ abstract class ApplicationMasterBase {
         config.setInteger(ConfigConstants.JOB_MANAGER_WEB_PORT_KEY, 0);
       }
 
-      val (actorSystem, jmActor, archiveActor, webMonitor) =
-        JobManager.startActorSystemAndJobManagerActors(
-          config,
-          JobManagerMode.CLUSTER,
-          streamingMode,
-          ownHostname,
-          0,
-          getJobManagerClass,
-          getArchivistClass
-        )
+      // we try to start the JobManager actor system using the port definition
+      // from the config.
+      // first, we check if the port is available by opening a socket
+      // if the actor system fails to start on the port, we try further
+      val amPortRange: String = config.getString(ConfigConstants.YARN_APPLICATION_MASTER_PORT,
+        ConfigConstants.DEFAULT_YARN_APPLICATION_MASTER_PORT)
+      val portsSet = NetUtils.getPortRangeFromString(amPortRange)
+
+      // Try to start the actor system with the given set of ports:
+      var startActorResultOption: Option[(ActorSystem, ActorRef, ActorRef, Option[WebMonitor])] =
+        None
+
+      while (startActorResultOption.isEmpty && portsSet.size() > 0) {
```

Comment:
Why is the `portsSet.size() > 0` condition included in the while loop condition. Does it change over time?